src/machines/java/com/enderio/machines/common/item/FluidTankItem.java

```diff
@@ -1,17 +1,40 @@
 package com.enderio.machines.common.item;
 
+import com.enderio.core.client.item.ItemTooltip;
 import com.enderio.machines.client.rendering.item.FluidTankBEWLR;
+import com.enderio.machines.common.block.MachineBlock;
 import net.minecraft.client.renderer.BlockEntityWithoutLevelRenderer;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.Component;
 import net.minecraft.world.item.BlockItem;
-import net.minecraft.world.level.block.Block;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraft.world.level.Level;
 import net.minecraftforge.client.extensions.common.IClientItemExtensions;
+import net.minecraftforge.common.capabilities.ForgeCapabilities;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
 import net.minecraftforge.common.util.NonNullLazy;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+import net.minecraftforge.fluids.capability.templates.FluidHandlerItemStack;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
+import java.util.List;
+import java.util.Optional;
 import java.util.function.Consumer;
 
 public class FluidTankItem extends BlockItem {
-    public FluidTankItem(Block block, Properties properties) {
+
+    protected final int capacity;
+    public FluidTankItem(MachineBlock block, Properties properties, int capacity) {
         super(block, properties);
+        this.capacity = capacity;
+    }
+
+    @Override
+    public int getMaxStackSize(ItemStack stack) {
+        return 1;//TODO: fix madness with tanks that stack.
```

Comment:
you can't fix stackable fluid containers, like just don't. dupe bugs everywhere because there is no common way to do this

Reply:
What I meant was that the EnderIO fluidtanks in 1.12 were stackable, and the current implementation only supports fluidtanks that doesn't stack because of the code in [FluidTankEntity].internalDrain/fill. The same bug will be apparent for fluid cells from forestry for example. But I'll move this comment to those methods.

Reply:
Or this was maybe what you meant? That we can't make them stack anymore?

Reply:
we technically can, but I've heard about a lot of weird interactions with stackable fluid containers over the years, it's probably not worth the bugs imo